### ISSUE TITLE
Add possibility to add sidecar containers to the main-container `redpanda`

### ISSUE THREAD

**Issue by GezimSejdiu** [redpanda-data/helm-charts#621]
Hi there, 

I'm trying to attach another sidecar container to the `redpanda` container and based on the crd: https://github.com/redpanda-data/redpanda/blob/9ade95e13279ca8a8471586134af995269c7a017/src/go/k8s/config/crd/bases/redpanda.vectorized.io_clusters.yaml#L661-L662 it says `Sidecars is a list of sidecars run alongside redpanda container` -- but it seems it does restrict to only those which have been specified on the specification e.g. https://github.com/redpanda-data/redpanda/blob/9ade95e13279ca8a8471586134af995269c7a017/src/go/k8s/config/crd/bases/redpanda.vectorized.io_clusters.yaml#L664 .

I was wondering if we can e.g. provide it via a podTemplate (see this as an example of what Flink community did: https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/custom-resource/pod-template/ ) or similar approaches so that we can also expand the deployment which will allow us to attach e.g. sidecars containers (as an example could be to add a log aggregation, or metric collection). 

I wasn't able to make it so that I can attach such a container alogside the `rpkStatus` to the `redpanda` main container or maybe I'm missing something. I do not want to patch the manifest after its deployment :( but rather want to have it deployet via the same manifest. 

Thanks a lot for doing great work with redpanda.

Best regards,

**Comment by joejulian** [comment]
Moving to helm-charts as this will need implemented in the redpanda helm chart now.

**Comment by joejulian** [comment]
@alejandroEsc This is solved, right?

**Comment by alejandroEsc** [comment]
> @alejandroEsc This is solved, right?

No, we have the ability to add extra init containers at the moment. I can add the same for additional side cars
